src/live_restore/live_restore_state.c

```diff
@@ -0,0 +1,444 @@
+        break;
+    case WTI_LIVE_RESTORE_STATE_CLEAN_UP:
+        strcpy(state_strp, "WTI_LIVE_RESTORE_STATE_CLEAN_UP");
+        break;
+    case WTI_LIVE_RESTORE_STATE_COMPLETE:
+        strcpy(state_strp, "WTI_LIVE_RESTORE_STATE_COMPLETE");
+        break;
+    }
+}
+
+/*
+ * __live_restore_state_from_string --
+ *     Convert a string to its live restore state.
+ */
+static int
+__live_restore_state_from_string(
+  WT_SESSION_IMPL *session, char *state_str, WTI_LIVE_RESTORE_STATE *statep)
+{
+
+    if (strcmp(state_str, "WTI_LIVE_RESTORE_STATE_NONE") == 0)
+        *statep = WTI_LIVE_RESTORE_STATE_NONE;
+    else if (strcmp(state_str, "WTI_LIVE_RESTORE_STATE_LOG_COPY") == 0)
+        *statep = WTI_LIVE_RESTORE_STATE_LOG_COPY;
+    else if (strcmp(state_str, "WTI_LIVE_RESTORE_STATE_BACKGROUND_MIGRATION") == 0)
+        *statep = WTI_LIVE_RESTORE_STATE_BACKGROUND_MIGRATION;
+    else if (strcmp(state_str, "WTI_LIVE_RESTORE_STATE_CLEAN_UP") == 0)
+        *statep = WTI_LIVE_RESTORE_STATE_CLEAN_UP;
+    else if (strcmp(state_str, "WTI_LIVE_RESTORE_STATE_COMPLETE") == 0)
+        *statep = WTI_LIVE_RESTORE_STATE_COMPLETE;
+    else
+        WT_RET_MSG(session, EINVAL, "Invalid state string: '%s' ", state_str);
+
+    return (0);
+}
+
+/*
+ * __live_restore_get_state_from_file --
+ *     Read the live restore state from the on-disk file. If it doesn't exist we return NONE. The
+ *     caller must already hold the live restore state lock. This function takes a *non-live
+ *     restore* file system, for example the backing posix file system used when accessing the
+ *     source or destination directly.
+ */
+static int
+__live_restore_get_state_from_file(WT_SESSION_IMPL *session, WT_FILE_SYSTEM *fs,
+  const char *backing_folder, WTI_LIVE_RESTORE_STATE *statep)
+{
+    WT_DECL_RET;
+    WT_FILE_HANDLE *fh = NULL;
+    bool state_file_exists = false;
+
+    WT_DECL_ITEM(state_file_name);
+    WT_RET(__wt_scr_alloc(session, 0, &state_file_name));
+    WT_ERR(__wt_filename_construct(session, backing_folder, WTI_LIVE_RESTORE_STATE_FILE,
+      UINTMAX_MAX, UINT32_MAX, state_file_name));
+
+    WT_ERR(
+      fs->fs_exist(fs, (WT_SESSION *)session, (char *)state_file_name->data, &state_file_exists));
+    if (!state_file_exists)
+        *statep = WTI_LIVE_RESTORE_STATE_NONE;
+    else {
+        char state_str[128];
+
+        WT_ERR(fs->fs_open_file(fs, (WT_SESSION *)session, (char *)state_file_name->data,
+          WT_FS_OPEN_FILE_TYPE_REGULAR, WT_FS_OPEN_EXCLUSIVE, &fh));
+
+        wt_off_t file_size;
+        WT_ERR(fs->fs_size(fs, (WT_SESSION *)session, (char *)state_file_name->data, &file_size));
+        WT_ERR(fh->fh_read(fh, (WT_SESSION *)session, 0, (size_t)file_size, state_str));
+
+        WT_ERR(__live_restore_state_from_string(session, state_str, statep));
+    }
+
+err:
+    if (fh != NULL)
+        WT_TRET(fh->close(fh, &session->iface));
+
+    if (state_file_name != NULL)
```

Comment:
This check is not needed and should be removed.

Reply:
https://github.com/wiredtiger/wiredtiger/pull/11563/commits/47864a97414760ee25e3dc15caeac54122b746ed